Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -0,0 +1,349 @@
+		let reply = Document()
+		var error = bson_error_t()
+		if !mongoc_collection_update_one(
+			self._collection, filter.data, update.data, opts?.data, reply.data, &error) {
+			throw MongoError.commandError(message: toErrorString(error))
+		}
+		return UpdateResult(from: reply)
+	}
+
+	/**
+	 * Updates multiple documents matching the provided filter in this collection.
+	 *
+	 * - Parameters:
+	 *   - filter: A `Document` representing the match criteria
+	 *   - update: A `Document` representing the update to be applied to matching documents
+	 *   - options: Optional `UpdateOptions` to use when executing the command
+	 *
+	 * - Returns: The optional result of attempting to update multiple documents. If the write
+	 *            concern is unacknowledged, nil is returned
+	 */
+	@discardableResult
+	public func updateMany(filter: Document, update: Document, options: UpdateOptions? = nil) throws -> UpdateResult? {
+		let encoder = BsonEncoder()
+		let opts = try encoder.encode(options)
+		let reply = Document()
+		var error = bson_error_t()
+		if !mongoc_collection_update_many(
+			self._collection, filter.data, update.data, opts?.data, reply.data, &error) {
+			throw MongoError.commandError(message: toErrorString(error))
+		}
+		return UpdateResult(from: reply)
+	}
+
+	/**
+	 * Deletes a single matching document from the collection.
+	 *
+	 * - Parameters:
+	 *   - filter: A `Document` representing the match criteria
+	 *   - options: Optional `DeleteOptions` to use when executing the command
+	 *
+	 * - Returns: The optional result of performing the deletion. If the `WriteConcern` is
+	 *            unacknowledged, `nil` is returned.
+	 */
+	@discardableResult
+	public func deleteOne(_ filter: Document, options: DeleteOptions? = nil) throws -> DeleteResult? {
+		let encoder = BsonEncoder()
+		let opts = try encoder.encode(options)
+		let reply = Document()
+		var error = bson_error_t()
+		if !mongoc_collection_delete_one(
+			self._collection, filter.data, opts?.data, reply.data, &error) {
+			throw MongoError.commandError(message: toErrorString(error))
+		}
+		return DeleteResult(from: reply)
+	}
+
+	/**
+	 * Deletes multiple documents
+	 *
+	 * - Parameters:
+	 *   - filter: Document representing the match criteria
+	 *   - options: Optional `DeleteOptions` to use when executing the command
+	 *
+	 * - Returns: The optional result of performing the deletion. If the `WriteConcern` is
+	 *            unacknowledged, `nil` is returned.
+	 */
+	@discardableResult
+	public func deleteMany(_ filter: Document, options: DeleteOptions? = nil) throws -> DeleteResult? {
+		let encoder = BsonEncoder()
+		let opts = try encoder.encode(options)
+		let reply = Document()
+		var error = bson_error_t()
+		if !mongoc_collection_delete_many(
+			self._collection, filter.data, opts?.data, reply.data, &error) {
+			throw MongoError.commandError(message: toErrorString(error))
+		}
+		return DeleteResult(from: reply)
+	}
+}
+
+/// Write command options structs
```

Comment:
If `///` is used for documentation, what happens to a floating comment like this? Does it get attached to the subsequent element (`InsertOneOptions` in this case)?